### PR TITLE
Add `customAttributes` input to `allPublishedContent`

### PR DESCRIPTION
**Pull request by Shinsina**
This is being utilized by the new custom redirect handler on RR here: https://github.com/parameter1/randall-reilly-websites/pull/315

Screen Shot 2021-11-30 at 10 16 32 AM

### CODE REVIEW

**Review by zarathustra323 — CHANGES_REQUESTED** [review]
Change this to query by custom attributes instead

**Comment by Shinsina** [comment]
[Add custom redirect handler for /?p=##](https://trello.com/c/1hluktAs/97-add-custom-redirect-handler-for-p)

File: services/graphql-server/src/graphql/definitions/platform/content/index.js [hunk 1]
```diff
@@ -485,6 +485,8 @@ input AllPublishedContentQueryInput {
   sectionId: Int
   "Deprecated. Use includeContentTypes instead."
   contentTypes: [ContentType!] = []
+  "Limit results to items matching specific custom attribute key value pairs"
+  customAttributes: [ContentCustomAttributeQueryInput] = []
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/index.js]: The custom attribute input (inside the array) should be non-nullable, i.e.
`[ContentCustomAttributeQueryInput!] = []`

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 2]
```diff
@@ -874,6 +875,11 @@ module.exports = {
         excludeContentTypes,
       });
 
+      customAttributes.forEach(({ key, value }) => {
+        const fullKeyPath = `customAttributes.${key}`;
+        query.$and.push({ [fullKeyPath]: value });
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: I would just use the template literal directly:

```js
query.$and.push({ [`customAttributes.${key}`]: value });
```

**Review by zarathustra323 — CHANGES_REQUESTED** [review]

**Review by zarathustra323 — APPROVED** [review]

**Pull request merged**